[Edit] SQL: COUNT() (#1707)
Codecademy/docs: content/sql/concepts/aggregate-functions/terms/count/count.md

FROM table_name;
```

> `COUNT()` takes the name of a column as an argument and counts the number of rows where the column is not `NULL`. The non `NULL` values are counted in the column that passed in using the `column_name`.

Reviewer: ```suggestion
`COUNT()` takes the name of a column as an argument and counts the number of rows where the column is not `NULL`. The non `NULL` values are counted in the column that is passed in using the `column_name`.
```

> When `*` or `1` are passed as an argument into `COUNT()`, the number of all rows, not excluding `NULL`, are included in the count.

Reviewer: ```suggestion
When `*` or `1` are passed as an argument into `COUNT()`, the number of all rows, including `NULL` values, are included in the count.
```

`COUNT()` is a function that takes the name of a column as an argument and counts the number of rows where the column is not `NULL`.

> **Note:** There are different variations to count records.  `COUNT(*)` counts number of rows including `NULL` values. `COUNT(1)` counts non `NULL` values. `COUNT(*)` and `COUNT(1)` have the same results because `COUNT(1)` replaces all records with the value `1` and counts the number of value `1`. The `COUNT(column_name)` counts non `NULL` values in that column.

Reviewer 2: What does COUNT(1) replace with the value 1 (Please include this)?

Can we also refer to the variations as what's passed in (like * or 1 rather than the entire COUNT() func) and reword accordingly.

Let's also take this out of a note and place the points in bullet format (line 26 can be part of these bullet points).

> **Note:** There are different variations to count records.  `COUNT(*)` counts number of rows including `NULL` values. `COUNT(1)` counts non `NULL` values. `COUNT(*)` and `COUNT(1)` have the same results because `COUNT(1)` replaces all records with the value `1` and counts the number of value `1`. The `COUNT(column_name)` counts non `NULL` values in that column.

Author: @yangc95 
Thank you for your feedback.  

I'm not sure if I understand your question: 
> What does COUNT(1) replace with the value 1 (Please include this)
Could you please clarify this?

I reordered examples and updated with bullet format.

> **Note:** There are different variations to count records.  `COUNT(*)` counts number of rows including `NULL` values. `COUNT(1)` counts non `NULL` values. `COUNT(*)` and `COUNT(1)` have the same results because `COUNT(1)` replaces all records with the value `1` and counts the number of value `1`. The `COUNT(column_name)` counts non `NULL` values in that column.

Reviewer 2: Yeah! Does it replace null values with 1?

> **Note:** There are different variations to count records.  `COUNT(*)` counts number of rows including `NULL` values. `COUNT(1)` counts non `NULL` values. `COUNT(*)` and `COUNT(1)` have the same results because `COUNT(1)` replaces all records with the value `1` and counts the number of value `1`. The `COUNT(column_name)` counts non `NULL` values in that column.

Author: @yangc95 
Yes!  Thank you for your help, I see what you mean.
It replaces all values including null with 1, so it counts all records including nulls.
I'll clarify that in the document.  Thanks!

> **Note:** There are different variations to count records.  `COUNT(*)` counts number of rows including `NULL` values. `COUNT(1)` counts non `NULL` values. `COUNT(*)` and `COUNT(1)` have the same results because `COUNT(1)` replaces all records with the value `1` and counts the number of value `1`. The `COUNT(column_name)` counts non `NULL` values in that column.

Author: @yangc95 
I just modified the document and pushed the sql-count-branch.  Could you please review it when you have a chance? Thanks!

| Michael | 80000  | 14         |
| Pam     | 41500  | 2          |
| Jim     | 45000  | 4          |
| Dwight  | 55000  | NULL       |

* 'COUNT()'
`COUNT()` is a function that takes the name of a column as an argument and counts the number of rows where the column is not `NULL`.

> The `COUNT(column_name)` counts non `NULL` values in that column.

Author: @yangc95 
I just pushed the edit. Could you please review this when you have a chance?

To find the total number of rows in the `employees` table that has experience, the given query can be used:

```sql
SELECT COUNT("experience")
FROM employees;
```

The result would be:

| COUNT("experience") |
| --------- |
| 3         |

* COUNT(\*) and COUNT(1)

> When * or 1 are used as an argument of 'COUNT()', that counts the number of all rows including `NULL`.

Author: @yangc95 
I just pushed the edit. Could you please review this when you have a chance?

| Michael | 80000  | 14         |
| Pam     | 41500  | 2          |
| Jim     | 45000  | 4          |
| Dwight  | 55000  | NULL       |

> To find the total number of rows in the `employees` table that has experience, the given query can be used:

Reviewer: ```suggestion
To find the total number of rows in the `employees` table that have experience, the given query can be used:
```

| Michael | 80000  | 14         |
| Pam     | 41500  | 2          |
| Jim     | 45000  | 4          |
| Dwight  | 55000  | NULL       |

To find the total number of rows in the `employees` table, the given query can be used:

```sql
SELECT COUNT(*)
FROM employees;
```

The result would be:

| COUNT(\*) |
| --------- |
| 4         |

> ```sql

Reviewer 2: Since the result is the same for both queries, can we group the two sql blocks together or list them one after another with some text to explain they will result in the same output?

Author: @yangc95 
Thank you for your feedback.  I combined the examples and showed only one result.

| COUNT(\*) |
| --------- |
| 2         |

> **Note:** NULL value is excluded due to the `WHERE experience < 5` command. 

Reviewer 2: Let's take this out of a note since it's not off topic to the example above.

Author: @yangc95 
Thank you for your feedback.  I removed this example from the document.
